invenio_app_ils/records/api.py

```diff
@@ -61,8 +65,32 @@ def create(cls, data, id_=None, **kwargs):
                 document_pid=data[cls.pid_field],
             )
         }
+        data["keywords"] = {
+            "$ref": cls._keyword_resolver_path.format(
+                scheme=current_app.config["JSONSCHEMAS_URL_SCHEME"],
+                host=current_app.config["JSONSCHEMAS_HOST"],
+                document_pid=data[cls.pid_field],
+            )
+        }
+        data.setdefault("keyword_pids", [])
         return super(Document, cls).create(data, id_=id_, **kwargs)
 
+    def add_keyword(self, keyword):
+        """Add a new keyword to the document."""
+        if keyword["keyword_pid"] not in self["keyword_pids"]:
+            self["keyword_pids"].append(keyword["keyword_pid"])
+
+    def remove_keyword(self, keyword):
+        """Remove a keyword from the document.
+
+        :returns: True if keyword was removed
+        """
+        try:
+            self["keyword_pids"].remove(keyword["keyword_pid"])
+        except ValueError:
+            return False
```

Comment:
maybe we could have an exception here to put a notification in the UI if it was not removed

Reply:
Yes, that would be better